### PR TITLE
Return full execution data on every request if under max specified size

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
This change returns fully populated input & output data on GetExecutionData requests in addition to signed URLs.

Not to be merged until https://github.com/lyft/flyteidl/pull/73 is checked in & released

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
This is useful to return inputs & outputs in sandbox mode when the signed url is host-specific. Also fixes the leaky abstraction presented by requiring a caller to concern themselves with where the data stored and having to parse the proto on their own.

## Tracking Issue
https://github.com/lyft/flyte/issues/419

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: pkg/manager/impl/execution_manager.go [hunk 1]
```diff
@@ -1019,10 +1021,32 @@ func (m *ExecutionManager) GetExecutionData(
 	if err != nil {
 		return nil, err
 	}
-	return &admin.WorkflowExecutionGetDataResponse{
+	response := &admin.WorkflowExecutionGetDataResponse{
```
Comment (kumare3) [inline pkg/manager/impl/execution_manager.go:1024]: @katrogan one reason to do this is to avoid having to sign the URL, so can we just do a `HEAD`, check the size and if that is ok then do a protobuf get using flytestdlib.storage?

Comment (kumare3) [reply pkg/manager/impl/execution_manager.go:1024]: or maybe i am misunderstanding, do we actually store the size of the payload in our database?

Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:1024]: until the UI transitions to using the new field, we should return both for backwards compatibility 
but good point on using flytestdlib, will update

Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:1024]: no, we don't store the size

Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:1024]: oh wait, and the storage client is using flytestdlib already
